web/src/components/dashboard/SidebarCardFundingNudge.test.tsx

```diff
@@ -1,29 +1,65 @@
 import { SidebarCardFundingNudge } from "@/components/dashboard/SidebarCardFundingNudge";
 import { getMergedConfig } from "@/contexts/configContext";
+import * as api from "@/lib/api-client/apiClient";
 import { SidebarCardContent } from "@/lib/types/types";
-import { generateSidebarCardContent } from "@/test/factories";
+import {
+  generateEmptyFormationData,
+  generateSidebarCardContent,
+  randomPublicFilingLegalType,
+} from "@/test/factories";
 import { selectDropdownByValue } from "@/test/helpers/helpers-testing-library-selectors";
+import { randomElementFromArray } from "@/test/helpers/helpers-utilities";
 import { useMockRouter } from "@/test/mock/mockRouter";
 import {
   currentBusiness,
   setupStatefulUserDataContext,
   userDataWasNotUpdated,
   WithStatefulUserData,
 } from "@/test/mock/withStatefulUserData";
-import { OperatingPhaseId } from "@businessnjgovnavigator/shared/";
+import {
+  FormationData,
+  FormationLegalType,
+  getCurrentBusiness,
+  OperatingPhaseId,
+  OperatingPhases,
+  UserData,
+} from "@businessnjgovnavigator/shared";
 import { SIDEBAR_CARDS } from "@businessnjgovnavigator/shared/domain-logic/sidebarCards";
 import {
   generateBusiness,
+  generateFormationData,
+  generateGetFilingResponse,
   generatePreferences,
   generateProfileData,
+  generateTaxFilingData,
   generateUserDataForBusiness,
+  randomLegalStructure,
 } from "@businessnjgovnavigator/shared/test";
 import { Business } from "@businessnjgovnavigator/shared/userData";
 import { fireEvent, render, screen } from "@testing-library/react";
 
 jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
 jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+jest.mock("@/lib/api-client/apiClient", () => ({
+  postTaxFilingsOnboarding: jest.fn(),
+  postTaxFilingsLookup: jest.fn(),
+}));
+
 jest.mock("next/router", () => ({ useRouter: jest.fn() }));
+const mockApi = api as jest.Mocked<typeof api>;
+
+const mockApiResponse = (userData: UserData, overrides: Partial<Business>): void => {
+  mockApi.postTaxFilingsOnboarding.mockResolvedValue({
+    ...userData,
+    businesses: {
+      ...userData.businesses,
+      [userData.currentBusinessId]: {
+        ...userData.businesses[userData.currentBusinessId],
+        ...overrides,
+      },
+    },
+  });
+};
```

Comment:
[boulder] This is great that we're setting this up, but it's not really adding anything. Commenting out this mock response doesn't impact how these tests work. What do we actually want to assert here?

It might be worth adding something in to confirm that we actually hit the mockAPI.